### PR TITLE
add tests for HardwareTypesTable component

### PR DESCRIPTION
**Pull request by harisbikovic**
This PR adds tests for the HardwareTypesTable component.

### CODE REVIEW

File: frontend/src/components/HardwareTypesTable.test.tsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  `;
+
+  const ComponentWithQuery = () => {
+    const { hardwareTypes } =
+      useLazyLoadQuery<HardwareTypesTable_getHardwareTypes_Query>(
+        GET_HARDWARE_TYPES_QUERY,
+        {},
+      );
+    return (
+      hardwareTypes && <HardwareTypesTable hardwareTypesRef={hardwareTypes} />
```
Comment (szakhlypa) [inline frontend/src/components/HardwareTypesTable.test.tsx:50]: Do we need to check on `hardwareTypes`?

File: frontend/src/components/HardwareTypesTable.test.tsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+  };
+  const SuspenseWithComponent = ({
+    fallback = "Loading...",
+  }: SuspenseWithComponentProps) => (
+    <Suspense fallback={fallback}>
+      <ComponentWithQuery />
+    </Suspense>
+  );
+
+  it("does suspension correctly", () => {
```
Comment (szakhlypa) [inline frontend/src/components/HardwareTypesTable.test.tsx:65]:
```suggestion
  it("suspends correctly", () => {
```

File: frontend/src/components/HardwareTypesTable.test.tsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    renderComponent([hwType1, hwType2, hwType3]);
+
+    const table = await screen.findByRole("table");
+    expect(table).toBeInTheDocument();
+    const tableElements = within(table).getAllByRole("rowgroup");
+    expect(tableElements).toHaveLength(2);
+
+    const [head, body] = tableElements;
+
+    const columns = within(head).getAllByRole("columnheader");
```
Comment (szakhlypa) [inline frontend/src/components/HardwareTypesTable.test.tsx:153]: Why do we repeat the `columnheader` check in every rendering test? Could we create separate unit-test for it?

File: frontend/src/components/HardwareTypesTable.test.tsx [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    expect(cells).toHaveLength(3);
+    expect(
+      within(cells[0]).getByRole("link", { name: hwType.name }),
+    ).toBeInTheDocument();
+    expect(cells[0]).toHaveTextContent(hwType.name);
+    expect(cells[1]).toHaveTextContent(hwType.handle);
+    expect(cells[2]).toHaveTextContent(hwType.partNumbers.join(", "));
+  });
+
+  it("renders table with multiple rows when data fetched contains multiple elements", async () => {
```
Comment (szakhlypa) [inline frontend/src/components/HardwareTypesTable.test.tsx:123]: What the reason to add this test? Rendering multiple rows should be tested on `Table.tsx` level.

File: frontend/src/components/HardwareTypesTable.test.tsx [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+    const row = rows[0];
+    const cells = within(row).getAllByRole("cell");
+    expect(cells).toHaveLength(3);
+    expect(
+      within(cells[0]).getByRole("link", { name: hwType.name }),
+    ).toBeInTheDocument();
+    expect(cells[0]).toHaveTextContent(hwType.name);
+    expect(cells[1]).toHaveTextContent(hwType.handle);
+    expect(cells[2]).toHaveTextContent(hwType.partNumbers.join(", "));
```
Comment (szakhlypa) [inline frontend/src/components/HardwareTypesTable.test.tsx:120]: We add a test to make sure that `HardwareType` with few `partNumbers`  has them rendered with comma separators, but there is no test for what happens when `HardwareType` has only one partNumber.

File: frontend/src/components/HardwareTypesTable.test.tsx [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+    const table = await screen.findByRole("table");
+    expect(table).toBeInTheDocument();
+    const tableElements = within(table).getAllByRole("rowgroup");
+    expect(tableElements).toHaveLength(2);
+
+    const [head, body] = tableElements;
+
+    const columns = within(head).getAllByRole("columnheader");
+    expect(columns).toHaveLength(3);
+    expect(columns[0]).toHaveTextContent("Hardware Type Name");
```
Comment (szakhlypa) [inline frontend/src/components/HardwareTypesTable.test.tsx:105]: I think it's better not to depend on the order of the columns and test them with 

```typescript
expect(screen.getByRole("columnheader", { name: "Hardware Type Name" })).toBeInTheDocument();
```
Same for the cells below.

Please let me know if you agree. Maybe we can have one test with the order check and all the others just checking for presence?

File: frontend/src/components/HardwareTypesTable.test.tsx [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+      handle: "HW-HANDLE3",
+      name: "HW name3",
+      partNumbers: ["HW-PN31", "HW-PN32"],
+    };
+    renderComponent([hwType1, hwType2, hwType3]);
+
+    const table = await screen.findByRole("table");
+    expect(table).toBeInTheDocument();
+    const tableElements = within(table).getAllByRole("rowgroup");
+    expect(tableElements).toHaveLength(2);
```
Comment (szakhlypa) [inline frontend/src/components/HardwareTypesTable.test.tsx:149]: Why do we repeat this code in every rendering test? Can we have one unit test for it?

File: frontend/src/components/HardwareTypesTable.test.tsx [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+
+    const rows = within(body).getAllByRole("row");
+    expect(rows).toHaveLength(1);
+
+    const row = rows[0];
+    const cells = within(row).getAllByRole("cell");
+    expect(cells).toHaveLength(3);
+    expect(
+      within(cells[0]).getByRole("link", { name: hwType.name }),
+    ).toBeInTheDocument();
```
Comment (szakhlypa) [inline frontend/src/components/HardwareTypesTable.test.tsx:117]: Do we want to test that link has correct href attribute?

File: frontend/src/components/HardwareTypesTable.test.tsx [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+    expect(cells[0]).toHaveTextContent(hwType.name);
```
Comment (szakhlypa) [inline frontend/src/components/HardwareTypesTable.test.tsx:118]: Expect seems redundant, you already tested on link's name just above.

**Review by szakhlypa — CHANGES_REQUESTED** [review]

File: frontend/src/components/HardwareTypesTable.test.tsx (hunk 6 above)
Comment (harisbikovic) [reply frontend/src/components/HardwareTypesTable.test.tsx:105]: Yes, I think that's a good idea.

**Review by szakhlypa — CHANGES_REQUESTED** [review]

**Review by szakhlypa — APPROVED** [review]

**Pull request merged**